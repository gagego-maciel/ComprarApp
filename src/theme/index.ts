export const theme = {
  colors: {
    primary: '#8257E5',
    background: '#121214',
    backgroundWhite: '#ffffff',
    backgroundSuccess: '#27ae60',
    backgroundButtonBlue: '#2c46b1',
    backgroundButtonRed: '#E53E3E',
    text: '#E1E1E6',
    textBlack: '#000000',
    textWhite: '#ffffff',
    textGray: '#A0A0A0',

    border: '#c3c5cb',

    placeHolderInput: '#74798B',
  },
}
