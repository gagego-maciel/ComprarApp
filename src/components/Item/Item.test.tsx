import React from 'react'
import { Item } from './index'
import { theme } from '@/theme'
import { ThemeProvider } from 'styled-components/native'
import { render, fireEvent } from '@testing-library/react-native'

jest.mock('@/assets/pendente-icon.png', () => 'pendente-icon')
jest.mock('@/assets/comprados-icon.png', () => 'comprados-icon')
jest.mock('@/assets/delete-icon.png', () => 'delete-icon')

const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}

describe('Item Component', () => {
  test('should render label correctly', () => {
    const { getByText } = renderWithTheme(
      <Item status="PENDING" label="Test item" />,
    )
    expect(getByText('Test item')).toBeTruthy()
  })

  test('should call onPressCompleted when pressing purchased button', () => {
    const onPressCompleted = jest.fn()
    const { getByTestId } = renderWithTheme(
      <Item
        status="PENDING"
        label="Item"
        onPressCompleted={onPressCompleted}
      />,
    )

    fireEvent.press(getByTestId('button-purchased'))
    expect(onPressCompleted).toHaveBeenCalled()
  })

  test('should call onPressRemoveItem when pressing delete button', () => {
    const onPressRemoveItem = jest.fn()
    const { getByTestId } = renderWithTheme(
      <Item
        status="PENDING"
        label="Item"
        onPressRemoveItem={onPressRemoveItem}
      />,
    )

    fireEvent.press(getByTestId('button-delete'))
    expect(onPressRemoveItem).toHaveBeenCalled()
  })

  test('should show line-through when status is PURCHASED', () => {
    const { getByText } = renderWithTheme(
      <Item status="PURCHASED" label="Completed item" />,
    )

    const label = getByText('Completed item')
    expect(label.props.style).toMatchObject({
      textDecorationLine: 'line-through',
    })
  })

  test('should not show line-through when status is PENDING', () => {
    const { getByText } = renderWithTheme(
      <Item status="PENDING" label="Pending item" />,
    )

    const label = getByText('Pending item')
    expect(label.props.style).toMatchObject({
      textDecorationLine: 'none',
    })
  })

  test('should matches snapshot', () => {
    const tree = renderWithTheme(
      <Item status="PENDING" label="Snapshot test" />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
