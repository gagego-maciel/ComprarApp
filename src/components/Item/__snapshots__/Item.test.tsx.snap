// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Component should matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderBottomColor": "#e4e6ec",
      "borderBottomWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <View
    style={{}}
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "gap": 5,
          "justifyContent": "center",
          "marginBottom": 16,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 16,
          "opacity": 1,
        }
      }
      testID="button-purchased"
    >
      <Image
        source="pendente-icon"
        style={
          {
            "height": 18,
            "width": 18,
          }
        }
      />
      <Text
        isPurchased="PENDING"
        style={
          {
            "fontSize": 14,
            "fontWeight": "500",
            "textDecorationLine": "none",
          }
        }
      >
        Snapshot test
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "marginBottom": 16,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 16,
        "opacity": 1,
      }
    }
    testID="button-delete"
  >
    <Image
      source="delete-icon"
      style={
        {
          "height": 18,
          "width": 18,
        }
      }
    />
  </View>
</View>
`;
